js/viz/core/errors_warnings.js

```diff
@@ -60,6 +60,14 @@ module.exports = errorUtils(errors.ERROR_MESSAGES, {
     */
     E2203: "The range you are trying to set is invalid",
     /**
+    * @name ErrorsViz_e2401
+    */
+    E2401: "A cycle detected in data that you provided",
+    /**
+    * @name ErrorsViz_e2402
+    */
+    E2402: "Provided data can not be displayed",
```

Comment:
Is there a reason behind this numbering? Although the grouping of errors by numbers may seem loose, some patterns can be found. For example, data related errors start with E20**. I suggest using E2006 and E2007 IDs.